rasa/shared/core/slots.py

```diff
@@ -27,16 +27,18 @@ class Slot:
     def __init__(
         self,
         name: Text,
+        mappings: List[Dict[Text, Any]],
         initial_value: Any = None,
         value_reset_delay: Optional[int] = None,
-        auto_fill: bool = True,
+        auto_fill: bool = False,
```

Comment:
Is it worth maybe leaving this as True? That would ensure existing code which uses the deprecated functionality gets the deprecation warning (which also changes the value to False, so we shouldn't have any issues correctness wise). I'm not sure how best to handle this - it looks like existing code should just work and leaving it set to False would mean new code written doesn't trigger the deprecation warning, but if the semantics are changing would it be worthwhile warning users? Is that maybe something we'd do after they've migrated to the new version on first run (excluding the migration guide)?